include/swift/AST/ASTContext.h

```diff
@@ -990,6 +990,13 @@ class ASTContext final {
   /// target triple.
   bool supportsVersionedAvailability() const;
 
+  /// Check's if our current availability context is contained within the given
+  /// feature's availability context. Prefer using this over the isContainedIn
+  /// function on availability context because this also checks if 1. we're
+  /// compiling the standard library (who has access to everything) and 2. if
+  /// we've disabled availability checking.
+  bool isAvailabilityAvailable(AvailabilityContext featureAvailability) const;
```

Comment:
Just double checking, is this only used to check for the availability of stdlib features, or is it also used to check for OS features? OS features still need to be conditionalized when building the stdlib, to support development at desk on OSes earlier than the very latest. Stdlib features obviously don't, so if that's all this is used for, then we're good.

Reply:
I think the documentation and name of this `ASTContext` API should somehow reflect that the result is typically based on the deployment target for normal modules. The fact that we build stdlib modules with an artificially low deployment target is mostly just a bug that we should rectify at some point.

And I think Mike's right that this maybe doesn't go far enough. Maybe we should explicitly model whether a feature is a Swift runtime/stdlib feature (always available regardless of deployment target) vs an OS feature.